app/security/WebAuthnCredentialRepository.scala

```diff
@@ -0,0 +1,122 @@
+package security
+
+import models.user.{MultiUserDAO, WebAuthnCredential, WebAuthnCredentialDAO}
+import com.yubico.webauthn._
+import com.yubico.webauthn.data._
+import com.scalableminds.util.accesscontext.GlobalAccessContext
+import com.scalableminds.util.objectid.ObjectId
+import com.typesafe.scalalogging.LazyLogging
+import net.liftweb.common.{Empty, Failure, Full}
+
+import java.util.Optional
+import javax.inject.Inject
+import scala.annotation.nowarn
+import scala.jdk.CollectionConverters._
+
+object WebAuthnCredentialRepository {
+  def byteArrayToBytes(arr: ByteArray): Array[Byte] = arr.getBytes
+  def bytesToByteArray(bytes: Array[Byte]): ByteArray = new ByteArray(bytes)
+
+  def objectIdToByteArray(id: ObjectId): ByteArray = new ByteArray(id.toString.getBytes())
+  def byteArrayToObjectId(arr: ByteArray): ObjectId = new ObjectId(new String(arr.getBytes))
+}
+
+/*
+ * UserHandle => ObjectId
+ * Username => User's E-Mail address
+ */
+
+@nowarn("cat=deprecation")
+class WebAuthnCredentialRepository @Inject()(multiUserDAO: MultiUserDAO, webAuthnCredentialDAO: WebAuthnCredentialDAO)
+    extends CredentialRepository
+    with LazyLogging {
+  def getCredentialIdsForUsername(email: String): java.util.Set[PublicKeyCredentialDescriptor] = {
+    val result = for {
+      user <- multiUserDAO.findOneByEmail(email)(GlobalAccessContext).await("Java interop")
+      keys <- webAuthnCredentialDAO.findAllForUser(user._id)(GlobalAccessContext).await("Java interop")
+      creds = keys.map(key => {
+        PublicKeyCredentialDescriptor.builder().id(WebAuthnCredentialRepository.bytesToByteArray(key.keyId)).build()
```

Comment:
Something like `this.bytesToByteArray` doesnt work? or maybe even without the this? :thinking:?

Reply:
Just tested: the without this is working fine if you import this file itself :see_no_evil:. Try using `import security.WebAuthnCredentialRepository.bytesToByteArray`

Reply:
Same please for `objectIdToByteArray` and the other used functions :pray: